shared/src/test/java/org/odk/collect/shared/collections/CollectionExtensionsTest.kt

```diff
@@ -0,0 +1,43 @@
+package org.odk.collect.shared.collections
+
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.Test
+import org.odk.collect.shared.collections.CollectionExtensions.itemFromHashOf
+
+class CollectionExtensionsTest {
+
+    @Test
+    fun `itemFromHashOf returns item using hashcode as index`() {
+        assertThat(listOf("0", "1").itemFromHashOf(HashCode(0)), equalTo("0"))
+        assertThat(listOf("0", "1").itemFromHashOf(HashCode(1)), equalTo("1"))
+    }
+
+    @Test
+    fun `itemFromHashOf wraps around when hashcode is greater than index`() {
+        assertThat(listOf("0", "1").itemFromHashOf(HashCode(2)), equalTo("0"))
+        assertThat(listOf("0", "1").itemFromHashOf(HashCode(5)), equalTo("1"))
+    }
+
+    @Test
+    fun `itemFromHashOf works with negative hashcode`() {
+        assertThat(listOf("0", "1").itemFromHashOf(HashCode(-1)), equalTo("1"))
+    }
+}
+
+private data class HashCode(private val value: Int) {
```

Comment:
Why do you need this class it seems like test work without it?

Reply:
It's being used in all the above tests. Do you mean we could write the test without it? My goal here was to use a class with a settable `hashcode` to make it clearer what's going on (rather than reverse engineering the implementation of an existing class.

Reply:
I mean you implemented this class to make sure that for 0 hashcode returns 0 for 1 retruns 1 etc
but if you pass 0, 1, etc directly to `itemFromHashOf` the result will be the same because that's how hashcode works for integers. So if you remove this wrapper class your tests still pass.

Reply:
Ah right I didn't actually realize that `hashcode()` works like that for `Int`. Although, it makes a lot of sense! Thinking about it though, the problem using `Int` as the test value is that the test would pass with an incorrect implementation:

```kotlin
abs(any) % this.size
```

Reply:
I can't change `abs(any.hashCode()) % this.size` to` abs(any) % this.size` it causes `None of the following functions can be called with the arguments supplied. `
Did you change anything else?

Reply:
Oh, sorry that will be because you can't call `abs` with `Any`! The bad implementation that would pass the test would be:

```kotlin
abs(any as Int) % this.size
```

Reply:
Ok I see.